python/pyspark/ml/pipeline.py

```diff
@@ -168,3 +168,24 @@ def transform(self, dataset, params={}):
         for t in self.transformers:
             dataset = t.transform(dataset, paramMap)
         return dataset
+
+
+class Evaluator(object):
+    """
+    Base class for evaluators that compute metrics from predictions.
+    """
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def evaluate(self, dataset, params={}):
```

Comment:
Python cannot overload methods. So it should be both `paramMaps` and `paramMap`. I used `params` here.